題目
[Day 3: Try, Catch, and Finally](https://www.hackerrank.com/challenges/js10-try-catch-and-finally/problem)
舉例輸入
>"1234"

舉例輸出
>4321

舉例輸入
>Number(1234)

舉例輸出
>s.split is not a function
>1234

解析
這篇就是如果輸入的是字串，則反轉
若不是字串則輸出字串切割錯誤
並且無論如何輸出參數或反轉結果

...其實這我有稍微卡一下，現在想起來很白癡
因為一般我都習慣會把處理過的參數另外存
這樣子的話我在這題就沒辦法印出原參數了


```js
function reverseString(s) {
    try { 
        s=s.split('').reverse().join('')
    }
    catch{ 
        console.log('s.split is not a function')
    }
    finally {console.log(s)}
}
```

---

題目
[Day 3: Throw](https://www.hackerrank.com/challenges/js10-throw/problem)
舉例輸入
>3  
1  
2  
3  

舉例輸出
>YES  
YES  
YES  
舉例輸入2
>YES
Zero Error
YES

舉例輸出2
>YES  
YES  
YES  

解析
大於0輸出YES  
等於0拋出Zero Error  
小於0拋出Negative Error  
很簡單的IF  


```js
function isPositive(a) {
    if (a > 0) {
        return 'YES';
    } else if (a == 0) {
        throw new Error('Zero Error');
    } else {
        throw new Error('Negative Error');
    }
}
```

本日心得
怎麼感覺有點沒辦法在十天內結束??
該不會一天打三題?


